src/utils/paginationUtil.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * Generates links for pagination.
+ * @param limit Max number of items per page.
+ * @param currPage Current page number.
+ * @param count Total number of items.
+ * @param sourceRoute Source route of where pagination is applied.
+ */
+function generatePagination(limit: number, currPage: number, count: number, sourceRoute: string) {
+  const lastPage = Math.ceil(count / limit);
+  let next = null;
+  let prev = null;
+  let err;
+  if (currPage >= 1 && currPage <= lastPage) {
+    next = currPage >= lastPage ? null : generateRoute(limit, currPage + 1, sourceRoute);
+    prev = currPage <= 1 ? null : generateRoute(limit, currPage - 1, sourceRoute);
+  } else {
+    err = 'Page not found.';
+  }
+
+  const pagination = [];
+  for (let i = 1; i <= lastPage; i++) {
+    pagination.push(generateRoute(limit, i, sourceRoute));
+  }
+  return {
+    next: next,
+    prev: prev,
+    err: err,
+    pagination: pagination,
+  };
+}
+
+/**
+ * Generates links for pagination with additional queries.
+ * @param limit Max number of items per page.
+ * @param currPage Current page number.
+ * @param count Total number of items.
+ * @param sourceRoute Source route of where pagination is applied.
+ * @param queries Query string to be appended at the end of each pagination link.
+ */
+function generatePaginationWithQueries(
```

Comment:
Could combine `generatePagination` and `generatePaginationWithQueries` into a single function. Let the queries be an optional parameter, clients will then just use `generatePagination` function to do pagination. If need append queries, send the queries in as the last parameter.